pkg/machines/components/create-vm-dialog/Password.jsx

```diff
@@ -0,0 +1,127 @@
+                passwordQuality: "",
+                warningQuality: ""
+            });
+        }
+    }
+
+    handleWarnings() {
+        setTimeout(() => {
+            this.setState({ displayWarningQual: true });
+        }, 300);
+    }
+
+    validateQuality(password) {
+        const dfd = cockpit.defer();
+        cockpit
+                .spawn("/usr/bin/pwscore", { err: "message" })
+                .input(password)
+                .done(content => {
+                    const quality = parseInt(content, 10);
+                    if (quality === 0) {
+                        this.passwordQualityFail(messages.weakPassword);
+                        dfd.reject(new Error(messages.weakPassword));
+                    } else if (quality <= 33) {
+                        this.passwordQualityPass("weak", password);
+                        dfd.resolve("weak");
+                    } else if (quality <= 66) {
+                        this.passwordQualityPass("okay", password);
+                        dfd.resolve("okay");
+                    } else if (quality <= 99) {
+                        this.passwordQualityPass("good", password);
+                        dfd.resolve("good");
+                    } else {
+                        this.passwordQualityPass("excellent", password);
+                        dfd.resolve("excellent");
+                    }
+                })
+                .fail(ex => {
+                    this.passwordQualityFail(ex.message || messages.rejectedPassword);
+                });
+        return dfd.promise();
+    }
+
+    passwordQualityFail(message) {
+        this.setState({ passwordQuality: "weak", warningQuality: message });
+        this.props.onValueChanged(undefined);
+    }
+
+    passwordQualityPass(quality, password) {
+        this.setState({ passwordQuality: quality, warningQuality: "" });
+        this.props.onValueChanged(password);
+    }
+
+    render() {
+        const passwordInvalid = this.state.warningQuality && this.state.displayWarningQual;
+
+        return (
+            <>
+                <FormGroup validationState={passwordInvalid ? "error" : undefined}>
+                    <input
+                        id={this.props.id}
+                        type="password"
+                        className="form-control"
+                        onChange={e => this.handleChangePassword(e)}
+                        onBlur={() => this.handleWarnings()}
+                    />
+                </FormGroup>
+                <div
+                    className={`progress password-strength-meter ${this.state.passwordQuality}`}
+                >
+                    <div className="progress-bar" />
+                    <div className="progress-bar" />
+                    <div className="progress-bar" />
+                    <div className="progress-bar" />
+                </div>
+                {this.state.warningQuality !== "" && this.state.displayWarningQual && (
```

Comment:
Also you don't need to do `!== ""` . Empty string is evaluated as false, so having just `this.state.warningQuality && this.state.displayWarningQual` would bring same result.